src/frontend/src/optimizer/plan_node/generic_derive.rs

```diff
@@ -385,3 +385,29 @@ impl<PlanRef: GenericPlanRef> GenericPlanNode for Expand<PlanRef> {
         self.input.ctx()
     }
 }
+
+impl<PlanRef: GenericPlanRef> GenericPlanNode for Union<PlanRef> {
+    fn schema(&self) -> Schema {
+        self.inputs[0].schema().clone()
```

Comment:
Should we add the hidden column here?

Reply:
Probably not. The input schema has already contains the hidden column and we can find them by `source_col`.